styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -450,36 +434,6 @@ public WorkflowState workflowState(WorkflowId workflowId) throws IOException {
     return builder.build();
   }
 
-  private <T> T storeWithRetries(FnWithException<T, IOException> storingOperation) throws IOException {
-    int storeRetries = 0;
-
-    while (storeRetries < MAX_RETRIES) {
-      try {
-        return storingOperation.apply();
-      } catch (ResourceNotFoundException e) {
-        throw e;
-      } catch (DatastoreException | IOException e) {
-        if (e.getCause() instanceof ResourceNotFoundException) {
-          throw (ResourceNotFoundException) e.getCause();
```

Comment:
Hope `ResourceNotFoundException` will never be the cause of `DatastoreException` which is unfortunately a `RuntimeException`.